x-pack/plugins/alerting/server/alerts_client/lib/get_summarized_alerts_query.ts

```diff
@@ -310,9 +311,14 @@ export const getQueryByScopedQueries = ({
       return;
     }
 
-    const scopedQueryFilter = generateAlertsFilterDSL({
-      query: scopedQuery as AlertsFilter['query'],
-    })[0] as { bool: BoolQuery };
+    const scopedQueryFilter = generateAlertsFilterDSL(
+      {
+        query: scopedQuery as AlertsFilter['query'],
+      },
+      {
+        analyzeWildcard: true,
```

Comment:
I noticed this setting but it was on and it didn't seem to do anything. I looked at the code for `buildEsQuery` since this is what we use to convert the KQL to DSL, and it seems like it only supports this flag for lucene queries. So sadly I had to inject this field manually.

Reply:
I agree with @maryam-saeidi. We should respect the user's settings and take the value from the Kibana advanced settings. Unless for some reason we do not want users to override the setting.

<img width="1828" alt="Screenshot 2024-10-09 at 3 00 13 PM" src="https://github.com/user-attachments/assets/ec7dad53-6e06-4e82-992f-31ff277e588e">

Reply:
Please take a closer look at the description, it has nothing to do with KQL, its the lucene string parser

Reply:
Indeed 🙂. I thought about it because for the `query:allowLeadingWildcards`, which refers to Lucece, we respect it in the KQL bar of the maintenance window. I think it is okay not to respect it at the moment.

<img width="1414" alt="Screenshot 2024-10-24 at 9 28 53 AM" src="https://github.com/user-attachments/assets/c43ed4f9-e6ad-47cf-a43a-2c8b3ae666a9">